bin/cron/deliver_poste_messages

```diff
@@ -245,20 +245,37 @@ class Deliverer
     @templates[name] = Poste::Template.new IO.read(path), engine
   end
 
-  def parse_addresses(string)
-    string.split(', ').map do |user_string|
-      if user_string.include? '<'
-        user_string =~ /^([^<]*)<([^>]*)>$/
-        {name: $1.strip, email: $2}
-      else
-        {name: nil, email: user_string}
-      end
+  def parse_address_string(string)
+    # A comma without surrounding double-quotes isn't necessarily bad, since it
+    # might be like this:
+    #   friendly name, phd <email@code.org>
+    #
+    # Though in such a case, it should really be provided like this:
+    #   "friendly name, phd", <email@code.org>
+    #
+    # But we are alerting just in case a sender has attempted to provide multiple
+    # email addresses like this:
+    #   friendly name <email@code.org>, second person <email2@code.org>
+    #
+    # Because we deprecated support for this pattern, believing that we've never
+    # actually used it.
+    if /,(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)/ =~ string
+      Honeybadger.notify(
+        error_message: "Comma outside of surrounding double-quotes detected in address string: #{string}"
+      )
+    end
+
+    if string.include? '<'
+      string =~ /^([^<]*)<([^>]*)>$/
+      {name: $1.strip, email: $2}
```

Comment:
In this outcome, `$1.strip` may be a quoted string `"example"` or an unquoted string `example`. In the former case, we'll have stripped outside the quotes but not inside, which might be surprising behavior. `"  example  "`  Do you have a sense of whether it's important to normalize this?

Reply:
It's a good improvement, but we've lived without it for a few months now, so I'm okay with leaving it unresolved in this PR.